index.js

```diff
@@ -1,49 +1,28 @@
 // bounce tests
-const tlds = require('haraka-tld')
 const { SPF } = require('haraka-plugin-spf')
 const net_utils = require('haraka-net-utils')
+const crypto = require('node:crypto')
+const Address = require('address-rfc2821').Address
 
 exports.register = function () {
   this.load_bounce_ini()
   this.load_bounce_bad_rcpt()
-  this.load_host_list()
-  this.load_allowed_msgid_domains()
-
-  this.register_hook('mail', 'reject_all')
-  this.register_hook('rcpt_ok', 'bad_rcpt')
-  this.register_hook('data', 'single_recipient')
-  this.register_hook('data', 'bounce_spf_enable')
-  this.register_hook('data_post', 'empty_return_path')
-  this.register_hook('data_post', 'bounce_spf')
-  this.register_hook('data_post', 'non_local_msgid')
-}
-
-exports.load_bounce_bad_rcpt = function () {
-  const raw_list = this.config.get('bounce_bad_rcpt', 'list', () => {
-    this.load_bounce_bad_rcpt()
-  })
-
-  this.cfg.invalid_addrs = raw_list.map((n) => n.toLowerCase())
-}
-
-exports.load_host_list = function () {
-  const raw_list = this.config.get('host_list', 'list', () => {
-    this.load_host_list()
-  })
-
-  this.cfg.host_list = raw_list.map((n) => n.toLowerCase())
-}
-
-exports.load_allowed_msgid_domains = function () {
-  const raw_list = this.config.get(
-    'bounce_allowed_msgid_domains',
-    'list',
-    () => {
-      this.load_allowed_msgid_domains()
-    },
-  )
-
-  this.cfg.allowed_msgid_domains = raw_list.map((n) => n.toLowerCase())
+  this.load_bounce_whitelist()
+
+  if (this.cfg.reject.all_bounces) this.register_hook('mail', 'reject_all')
+  if (this.cfg.reject.bad_rcpt) this.register_hook('rcpt_ok', 'bad_rcpt')
+  if (this.cfg.check.single_recipient)
```

Comment:
The advantage of conditionally loading the hook here is that it saves Haraka from calling the hook on every connection. The disadvantage is that the function isn't enabled automatically when the config file is edited. You have to restart Haraka afterwards for the change to take effect. Since computing is so cheap, we usually opt to optimize for humans.

Reply:
Fixed